src/Service/MailchimpSyncContact.php

```diff
@@ -0,0 +1,166 @@
+        $this->httpClient = HttpClient::create([
+            'headers' => [
+                'Content-type' => 'application/json',
+                'Authorization' => [
+                    'Basic' => base64_encode(sprintf('key:%d', $this->container->getParameter('mailchimp.api_key'))),
+                ],
+            ],
+        ]);
+        $this->flashBag = $flashBag;
+        $this->mailer = $mailer;
+        $this->twig = $twig;
+    }
+
+    public function addContact(User $user)
+    {
+        $subscriptionStatus = $this->checkSubscriptionStatus($user);
+
+        if (!$subscriptionStatus || 404 === $subscriptionStatus) {
+            $isAdded = $this->requestApi(
+                $user,
+                'POST',
+                $this->container->getParameter('mailchimp.api_url'),
+                [
+                    'email_address' => $user->getEmail(),
+                    'status' => self::STATUS_SUBSCRIBED,
+                    'merge_fields' => [
+                        'FNAME' => $user->getName(),
+                        'LNAME' => $user->getName(),
+                    ],
+                ]
+            );
+        }
+    }
+
+    public function subscribe(User $user)
+    {
+        $this->requestApi(
+            $user,
+            'PUT',
+            $this->container->getParameter('mailchimp.api_url').'/'.md5(strtolower($user->getEmail())),
+            ['status' => self::STATUS_SUBSCRIBED]
+        );
+    }
+
+    public function unsubscribe(User $user)
+    {
+        $this->requestApi(
+            $user,
+            'PUT',
+            $this->container->getParameter('mailchimp.api_url').'/'.md5(strtolower($user->getEmail())),
+            ['status' => self::STATUS_UNSUBSCRIBED]
+        );
+    }
+
+    public function checkSubscriptionStatus(User $user)
+    {
+        return $this->requestApi(
+            $user,
+            'GET',
+            $this->container->getParameter('mailchimp.api_url').'/'.md5(strtolower($user->getEmail()))
+        );
+    }
+
+    private function requestApi(
+        User $user,
+        string $method,
+        string $url,
+        array $data = null,
+        bool $isNewContact = false
+    ) {
+        $options = [];
+        $expectedStatus = null;
+        if ($data) {
+            $options = ['body' => json_encode($data)];
+            if ($data['status']) {
+                $expectedStatus = $isNewContact ? self::STATUS_ADDED : $data['status'];
+            }
+        }
+
+        try {
+            $jsonContent = $this->httpClient->request(
+                $method,
+                $url,
+                $options
+            )
+                ->getContent();
+
+            $content = json_decode($jsonContent);
+            if (!empty($content) && !empty($content['status'])) {
+                return $content['status'];
+            }
+        } catch (TransportExceptionInterface | ClientExceptionInterface | RedirectionExceptionInterface | ServerExceptionInterface $e) {
```

Comment:
If there is no specific catching for `(TransportExceptionInterface | ClientExceptionInterface | RedirectionExceptionInterface | ServerExceptionInterface` why don't you just catche `Exception`?

Reply:
PhpStorm advise but you're right